src/test/java/com/wikia/webdriver/testcases/activityfeedstests/WikiActivityTests.java

```diff
@@ -73,97 +56,87 @@ public void WikiActivityTests_003_newBlogCreationIsRecordedOnActivityModule() {
+      String.format("Link in activities list for article with title %s "
+        + "redirected to article with title %s", title, article.getArticleTitle()));
   }
 
-  @Test(groups = "WikiActivity_007")
-  @Execute(asUser = User.USER)
-  public void WikiActivityTests_007_clickingUserRedirectsToUserPage() {
-    new ArticleContent().push("content");
-    new ArticleContent().push("content_after_edition");
+  public void clickingUsernameRedirectsToUserPage() {
+    Activity articleActivity = createArticleEditAndGetRecentActivity();
+
+    String expectedUserName = articleActivity.getUserLink().getText();
+    UserProfilePage userPage = articleActivity.clickOnUserLink();
 
-    SpecialWikiActivityPageObject activityPage = new SpecialWikiActivityPageObject(driver);
-    activityPage.open();
-    List<Activity> activityList = activityPage.getActivities(10);
+    Assertion.assertEquals(userPage.getUserName(), expectedUserName,
+      String.format("Link in activities list for username %s "
+        + "redirected to user profile for %s", expectedUserName, userPage.getUserName()));
+  }
 
-    Activity chosenActivity = activityList.stream().findFirst().get();
-    String expectedUserName = chosenActivity.getUser();
-    UserProfilePage userPage = chosenActivity.clickOnUserLink();
-    String currentUserName = userPage.getUserName();
+  public void clickingIconNextToArticleRedirectsToDiff() {
+    DiffPagePageObject diffPage = createArticleEditAndGetRecentActivity().clickOnDiffLink();
 
-    Assertion.assertEquals(currentUserName, expectedUserName);
+    Assertion.assertTrue(diffPage.isDiffTableVisible(),
+      "Diff table was not found on page");
   }
 
-  @Test(groups = "WikiActivity_008")
-  @Execute(asUser = User.USER)
-  public void WikiActivityTests_008_clickingIconNextToArticleRedirectsToDiff() {
-    new ArticleContent().push("content");
-    new ArticleContent().push("content_after_edition");
+  private Activity createArticleEditAndGetRecentActivity() {
+    ArticleContent content = new ArticleContent(testUser);
+    content.push("content");
+    content.push("content_after_edition");
 
-    SpecialWikiActivityPageObject activityPage = new SpecialWikiActivityPageObject(driver);
-    activityPage.open();
-    List<Activity> activityList = activityPage.getActivities(10);
-    EditActivity editActivity = (EditActivity) activityList
-        .stream()
-        .filter(activity -> activity instanceof EditActivity)
-        .findFirst().get();
+    return new SpecialWikiActivityPageObject()
+      .open()
+      .getMostRecentEditActivity();
+  }
 
-    DiffPagePageObject diffPage = editActivity.showChanges();
+  private ArticlePageObject createArticle() {
+    new ArticleContent(testUser).push("content");
+    return new ArticlePageObject().open();
+  }
 
-    Assertion.assertTrue(diffPage.isDiffTableVisible());
+  private String editArticleWithContentAndGetTitle(ArticlePageObject article, String content) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "editArticleWithContentAndGetTitle" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)